test/unit/range/decorator/gap_decorator_test.cpp

```diff
@@ -58,6 +57,76 @@ class aligned_sequence<gap_decorator<inner_type_>> : public ::testing::Test
 
 INSTANTIATE_TYPED_TEST_CASE_P(gap_decorator, aligned_sequence, test_types);
 
+using iterator_type = decltype(std::ranges::begin(std::declval<decorator_t &>()));
+using const_iterator_type = decltype(std::ranges::begin(std::declval<decorator_t const &>()));
+using random_access_iterator_type = decltype(std::declval<decorator_t &>().begin_ra());
+using const_random_access_iterator_type = decltype(std::declval<decorator_t &>().cbegin_ra());
+
+template <>
+struct iterator_fixture<iterator_type> : public ::testing::Test
+{
+    using iterator_tag = std::bidirectional_iterator_tag;
+    static constexpr bool const_iterable = true;
+
+    std::vector<dna4> const vec{"ACTGACTG"_dna4};
+    decorator_t dec = [this] ()
```

Comment:
Why do we need to capture the `this` pointer (as copy)? 
I don't see that it is used, also in `range_to_compare` below.

Reply:
I use the member `vec` inside both of the lambdas and since `this` is just a pointer the copy is cheap.

Reply:
Keep in mind that the rules for capturing this have change in c++20. Please build your code once with g++9 in std=c++2a mode to make sure it produces no warnings.

Reply:
How do I build with c++2 in our build system . Does simply setting the std flag override/adds it?

Reply:
Yes, plain cmake.

Reply:
Isn’t capturing this->vec the best solution then?

Reply:
ok then it works